src/core/ucc_progress_queue_mt.c

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+#include "ucc_progress_queue.h"
+#include "utils/ucc_malloc.h"
+#include "utils/ucc_log.h"
+#include "utils/ucc_spinlock.h"
+#include "utils/ucc_atomic.h"
+
+typedef struct ucc_pq_mt {
+    ucc_progress_queue_t super;
+    ucc_spinlock_t       locked_queue_lock;
+    ucc_coll_task_t***   tasks;
+    uint8_t              which_pool;
+    ucc_list_link_t      locked_queue;
+    uint32_t             tasks_countrs[2];
+} ucc_pq_mt_t; // TODO the struct isn't a queue because not maintaining order, maybe another name
+
+#define LINE_SIZE 8 // Could be changed, but in tests performed great
```

Comment:
plz give meaningful comments instead of this: like, LINE_SIZE 8 // number of tasks in a single lock free pool